lib/system/ctypes.nim

```diff
@@ -1,12 +1,22 @@
 ## Some type definitions for compatibility between different
 ## backends and platforms.
 
+when sizeof(int) == 8:
+  type
+    BiggestInt* = int64
+      ## is an alias for the biggest signed integer type the Nim compiler
+      ## supports.
+elif sizeof(int) == 4:
+  type
+    BiggestInt* = int32
```

Comment:
Well, that would depend on the underlying C compiler - those based on LLVM for example would emulate any number of bits (up to 64, given how Nim is limited to 64 bits and doesn't have larger native types) on any CPU.

If instead `BiggestInt` should represent the biggest integer that the *CPU* supports, it will again depend on hardware and CPU mode - 32-bit cpus typically have a 16-bit mode just like 64-bit cpus have a 32-bit mode, and "integers" and "pointers" on those addressing modes are not the same size - the integers are typically bigger and pointers are kept smaller to save memory - in Nim, `int` is the same size as the `pointer` so it becomes the case that the CPU supports a bigger integer natively than what the Nim `int` is: this is where using a type like `BiggestInt` makes sense: for performance, it usually makes sense to use the largest integer available on the CPU, specially when working with bigints / cryptography and so on.

If `BiggestInt` should follow the `int` size always, there's no point using it - might as well use `int` at that point.

Reply:
thanks! I think I get the point from your description, so it's like in the future if there's cpu support 128 bit it will map to int128